crates/goose-cli/src/recipe.rs

```diff
@@ -68,3 +69,44 @@ pub fn load_recipe<P: AsRef<Path>>(path: P, log: bool) -> Result<Recipe> {
 
     Ok(recipe)
 }
+
+fn render_string_with_params(
+    tera: &mut Tera,
+    content: &str,
+    context: &TeraContext,
+) -> Result<String> {
+    match tera.render_str(content, context) {
+        Ok(rendered) => Ok(rendered),
+        Err(_) => Err(anyhow::anyhow!(
+            "Failed to render the recipe - please check if all required parameters are provided"
+        )),
+    }
+}
+
+pub fn load_and_apply_recipe<P: AsRef<Path>>(
+    path: P,
+    log: bool,
+    params: Vec<(String, String)>,
+) -> Result<Recipe> {
+    let recipe = load_recipe(&path, log)?;
```

Comment:
So I think we should apply the template engine on the entire yaml file before we read it in as a yaml

You might want to do something like:

     goose run --recipe try_this --param temperature=0.3 
     
and then have it modify the temperature (once I merge my settings)